include/aikido/planner/ConfigurationToConfiguration.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef AIKIDO_PLANNER_CONFIGURATIONTOCONFIGURATION_HPP_
+#define AIKIDO_PLANNER_CONFIGURATIONTOCONFIGURATION_HPP_
+
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planning problem to plan to a single goal configuration.
+///
+/// Plan a trajectory from start state to goal state by using an interpolator to
+/// interpolate between them.
+class ConfigurationToConfiguration : public Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param stateSpace State space.
+  /// \param startState Start state.
+  /// \param goalState Goal state.
+  /// \param interpolator Interpolator used to produce the output trajectory.
+  /// \param constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw If \c stateSpace is not compatible to \c constraint's state space.
+  ConfigurationToConfiguration(
+      const statespace::ConstStateSpacePtr& stateSpace,
+      const statespace::StateSpace::State* startState,
+      const statespace::StateSpace::State* goalState,
+      statespace::InterpolatorPtr interpolator,
+      constraint::TestablePtr constraint);
+
+  // Documentation inherited.
+  const std::string& getName() const override;
+
+  /// Returns the name of the planner problem.
+  static const std::string& getStaticName();
+
+  /// Sets the start state.
```

Comment:
For Single Source Single Target, I agree that we _could_ create new problems. I don't see any problem with reusing by changing the start and goal though. For instance, at least the current implementation of `planToConfigurations` under `robot/util.cpp` is just iterating through each goal state and calling `planToConfiguration(goal)`. If we are doing that here as well, we could just `setGoalState` instead of creating a new instance everytime.